File: SYP.Web/Modules/ServerTypes/Setting/BankAccountInformationsEditorColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BankAccountInformationsRow } from "./BankAccountInformationsRow";

export interface BankAccountInformationsEditorColumns {
    Id: Column<BankAccountInformationsRow>;
    Firm: Column<BankAccountInformationsRow>;
    Bank: Column<BankAccountInformationsRow>;
    Branch: Column<BankAccountInformationsRow>;
    BranchCode: Column<BankAccountInformationsRow>;
    AccountNo: Column<BankAccountInformationsRow>;
    Iban: Column<BankAccountInformationsRow>;
    Swift: Column<BankAccountInformationsRow>;
    CurrencyCode: Column<BankAccountInformationsRow>;
    Origin: Column<BankAccountInformationsRow>;
    Payment: Column<BankAccountInformationsRow>;
    Shipment: Column<BankAccountInformationsRow>;
    TenantId: Column<BankAccountInformationsRow>;
}

export class BankAccountInformationsEditorColumns extends ColumnsBase<BankAccountInformationsRow> {
    static readonly columnsKey = 'Setting.BankAccountInformationsEditor';
    static readonly Fields = fieldsProxy<BankAccountInformationsEditorColumns>();
}

File: SYP.Web/Modules/Setting/BankAccountInformations/Editor/BankAccountInformationsEditorColumns.cs
namespace SYP.Setting.Columns;

[ColumnsScript("Setting.BankAccountInformationsEditor")]
[BasedOnRow(typeof(BankAccountInformationsRow), CheckNames = true)]
public class BankAccountInformationsEditorColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Firm { get; set; }
    public string Bank { get; set; }
    public string Branch { get; set; }
    public string BranchCode { get; set; }
    public string AccountNo { get; set; }
    public string Iban { get; set; }
    public string Swift { get; set; }
    [DisplayName("Para Birimi")]
    public string CurrencyCode { get; set; }
    public string Origin { get; set; }
    public string Payment { get; set; }
    public string Shipment { get; set; }
    public int TenantId { get; set; }
}